addons/web/static/src/js/fields/basic_fields.js

```diff
@@ -622,14 +622,46 @@ var FieldDateRange = InputField.extend({
         }
         this._super.apply(this, arguments);
     },
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Field widget is valid if value entered can convered to date/dateime value
+     * while parsing input value to date/datetime throws error then widget considered
+     * invalid
+     *
+     * @override
+     */
+    isValid: function () {
+        const value = this.mode === "readonly" ? this.value : this.$input.val();
+        try {
+            return field_utils.parse[this.formatType](value, this.field, { timezone: true }) || true;
+        } catch (error) {
+            return false;
+        }
+    },
```

Comment:
Why is this method never used ? Did you forgot some refactoring ?

Reply:
@SimonGenin It is internally called by framework, check here: https://github.com/odoo/odoo/blob/master/addons/web/static/src/legacy/js/views/basic/basic_renderer.js#L365